### DISCUSSION TITLE
Flipped order of config overrides

### DISCUSSION THREAD

**Discussion by TeriyakiBomb** [ghostty-org/ghostty#3820] [Feature Requests, Ideas] [answered]
I feel the order Ghostty prioritizes config files might make sense the other way around, at least on Mac OS. If I do not have ~/.config/ghostty/config created then as described in the docs it uses the config file in application support, as stated by the docs:

> If both locations exist, they are loaded in the order above with conflicting values in later files overriding earlier ones. Configuration is optional and if no configuration file is found, Ghostty will use its defaults.

But for a Mac user, admittedly anecdotally, you'd expect the config inside Library to be the default to be overridden. ~/.config is common for Mac OS developers and $XDG_CONFIG is probably a little less common. With the current order on Mac OS it requires you deleting the file in application support, which feels weird. For example, I manage my configs with chezmoi and the general flow would be:

- Play with some config in a default location or via CLI etc
- Move it to ~/.config and add to chezmoi when I want it to be portable

I imagine most users would expect a config created in ./config to override the more "default" location? Although I might be overlooking something.

Anyway, food for thought.

Also, I'm really enjoying Ghostty so far! It's just kind of ticking all of the boxes and I've been between terminal emulators for around a year now. So it might end up being the one I keep hold of going forward :)

**Comment by mitchellh** [comment, marked as answer]
Related: #3953 

The order of priority was well discussed before and has a reason: it is least-specific to most-specific. XDG dirs tend to be more cross-platform so it allows people to put shared config there and macOS specific config in AppSupport. Even if that's rare, the priority is easily explained as least-to-most specific.
